pyproject.toml

```diff
@@ -26,12 +26,13 @@ classifiers = [
     "Programming Language :: Python :: 3.8",
     "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
 ]
 dependencies = [
     "jupyter_server>=1.6,<3",
     "traitlets",
     "nbconvert",
-    "pydantic",
+    "pydantic>=1,<2",
```

Comment:
I would definitely leverage the compatible release operator `~=` as much as possible. The only quirk of this operator is that it behaves differently depending on how many version components you specify in the operand:

- `~=1.2` == `>=1.2.0,<2.0.0`
- `~=1.2.3` == `>=1.2.3,<1.3.0`

Here, we want `pydantic~=1.10`.